src/Common/RemoteProxyConfigurationResolver.cpp

```diff
@@ -16,20 +16,63 @@ namespace ErrorCodes
     extern const int BAD_ARGUMENTS;
 }
 
+std::string RemoteProxyHostFetcherImpl::fetch(const Poco::URI & endpoint, const ConnectionTimeouts & timeouts) const
+{
+    /// It should be just empty GET request.
+    Poco::Net::HTTPRequest request(Poco::Net::HTTPRequest::HTTP_GET, endpoint.getPath(), Poco::Net::HTTPRequest::HTTP_1_1);
+
+    const auto & host = endpoint.getHost();
+    auto resolved_hosts = DNSResolver::instance().resolveHostAll(host);
+
+    HTTPSessionPtr session;
+
+    for (size_t i = 0; i < resolved_hosts.size(); ++i)
```

Comment:
Do you intended to do retries? Why is it depend on IP's count in resolve result?
I think it is better use here some our common class which does requests with good retries.